### DISCUSSION TITLE
Localized Date, Datetime, Time, and Number

### DISCUSSION THREAD

**Discussion by kulturbande** [sveltekit-i18n/lib#62] [Q&A] [answered]
I'm currently evaluating this lib as a i18n - solution and I have quit offen the case, that I want to localize a date or a number. I'm currently solving this with an additional store, like this one:

```typescript
export const datetime = derived(locale, ($locale) => (date, option: Intl.DateTimeFormatOptions = {}) => {
    if (!(date instanceof Date)) {
        date = new Date(date);
    }
    return new Intl.DateTimeFormat($locale, option).format(date);
});
``` 

and call it in Svelte:
```svelte
<span>{$datetime(createdAt)}</span>
```

Are you likly to add these functionality? If yes, I could also provide one (or multiple) Pull Requests.

**Comment by jarda-svoboda** [comment, marked as answer]
Hi @kulturbande!

This functionality is already included in [@sveltekit-i18n/parser-default](https://github.com/sveltekit-i18n/parsers/blob/e95e7bbe38df66dab069d1ec7049fb0d30a50ee7/parser-default/src/modifiers.ts#L32-L49).

As you can see in the [parser-default example](https://github.com/sveltekit-i18n/lib/tree/master/examples/parser-default) you can use [modifiers](https://github.com/sveltekit-i18n/parsers/blob/master/parser-default/README.md#modifiers) (defined [here](https://github.com/sveltekit-i18n/lib/blob/b132fb6509f04906f4e222da2a42508ab13483bb/examples/parser-default/src/lib/translations/en/content.json#L16-L17) and used [here](https://github.com/sveltekit-i18n/lib/blob/b132fb6509f04906f4e222da2a42508ab13483bb/examples/parser-default/src/routes/index.svelte#L48)) to get what you need.) If build-in modifiers do not fit your needs, you can even [create your own](https://github.com/sveltekit-i18n/lib/blob/b132fb6509f04906f4e222da2a42508ab13483bb/examples/parser-default/src/lib/translations/modifiers.ts)!:)

You can use either [`@sveltekit-i18n/base`](https://github.com/sveltekit-i18n/base) together with [`@sveltekit-i18n/parser-default`](https://github.com/sveltekit-i18n/parsers/tree/master/parser-default) (as presented in the example) or simply [`sveltekit-i18n`](https://github.com/sveltekit-i18n/lib) which basically glues these two packages together.

Hope it will help.)

PS: If you'd rather use [ICU message syntax](https://unicode-org.github.io/icu/userguide/format_parse/messages/) take a look at [`@sveltekit-i18n/parser-icu`](https://github.com/sveltekit-i18n/parsers/tree/master/parser-icu) – [example here](https://github.com/sveltekit-i18n/lib/tree/master/examples/parser-icu) – it provides `number` and `date` formatters as well.

**Reply by kulturbande** [reply]
I do like the idea of modifiers (a little bit like pipes in Angular). In my case the values are in a table without further content (only the value itself). But I can do what I want. Maybe I go with the modifieres, maybe I'm writing my own derived stores. 

I do like your lib. Especially the interface, that the stores e.g. `$t` are pointing to a project owned translation file.

[a comment by an automated account is not shown]
